src/gui/modalMenu.cpp

```diff
@@ -273,7 +273,7 @@ bool GUIModalMenu::preprocessEvent(const SEvent &event)
 		irr_ptr<GUIModalMenu> holder;
 		holder.grab(this); // keep this alive until return (it might be dropped downstream [?])
 
-		if (event.TouchInput.ID == 0) {
```

Comment:
ouch, this shouldn't have passed code review. easy to miss however.

It sounds like what the code should be doing is remember the ID of the first and second touch when they begin.

Reply:
I did not realise that the ID assignment is different for each device. I just checked on my phone and thought that it resets to zero when there is no touch. My bad.